docs/python/debugging.md

```diff
@@ -55,11 +55,223 @@ Selecting the **Node.js: Gulp task** yields the following result:
+    "type": "python",
+    "request": "attach",
+    "host": "remote-machine-name", // replace this with remote machine name
+    "port": 5678,
+}
+```
+
+> **Note**: Be aware that when you specify a host value other than `127.0.0.1` or `localhost` you are opening a port to allow access from any machine, which carries security risks. You should make sure that you're taking appropriate security precautions, such as using SSH tunnels, when doing remote debugging.
+
+|Flag  |Options  |Description  |
+|---------|---------|---------|
+|**--listen** or **--connect**  |  `[<host>:]<port>`       |   **Required**. Specifies the host address and port for the debug adapter server to wait for incoming connections (--listen) or to connect with a client that is waiting for an incoming connection (--connect). This is the same address that is used in the VS Code debug configuration. By default the host address is `localhost (127.0.0.1)`.      |
+|**--wait-for-client**     |   none      | **Optional**. Specifies that the code should not run until there's a connection from the debug server. This setting allows you to debug from the first line of your code.        |
+|**--log-to**     |   `<path>`      | **Optional**. Specifies a path to an existing directory for saving logs.         |
+|**--log-to-stderr**     |    none     |  **Optional**. Enables debugpy to write logs directly to stderr.       |
+|**--pid**     |    `<pid>`     | **Optional**. Specifies a process that is already running to inject the debug server into.        |
+|**--configure-<name>** | `<value>` | **Optional**. Sets a debug property that must be known to the debug server before the client connects. Such properties can be used directly in *launch* configuration, but must be set in this manner for *attach* configurations. For example, if you don't want the debug server to automatically inject itself into subprocesses created by the process you're attaching to, use `--configure-subProcess false`.|
+
+> **Note**: `[<arg>]` can be used to pass command line arguments along to the app being launched.
+
+## Debugging by attaching over a network connection
+
+### Local script debugging
+
+In some scenarios, you need to debug a Python script that's invoked locally by another process. For example, you may be debugging a web server that runs different Python scripts for specific processing jobs. In such cases, you need to attach the VS Code debugger to the script once it's been launched:
+
+1. Run VS Code, open the folder or workspace containing the script, and create a `launch.json` for that workspace if one doesn't exist already.
+
+1. In the script code, add the following and save the file:
+
+    ```python
+    import debugpy
+
+    # 5678 is the default attach port in the VS Code debug configurations. Unless a host and port are specified, host defaults to 127.0.0.1
+    debugpy.listen(5678)
+    print("Waiting for debugger attach")
+    debugpy.wait_for_client()
+    debugpy.breakpoint()
```

Comment:
```suggestion
    debugpy.breakpoint()
    print('break on this line')
```
/cc @int19h we need the `print` right. Since we break on the line that we are about to execute?

Reply:
Never mind. This is what we are recommend people add to their files. It will break on the next line we will execute. This is only a issue if there are no more valid lines after `debugpy.breakpoint()`.

Reply:
Sounds like maybe we do need it then? You're saying if the file was just the code block in this snippet it wouldn't break right, since it stops on the line after the debugpy.breakpoint() call?

Reply:
If it is just the above code block then we will need the print line.